lib/foreman_ansible_core.rb

```diff
@@ -21,10 +21,13 @@ module ForemanAnsibleCore
 
   require 'foreman_remote_execution_core/actions'
   require 'foreman_ansible_core/remote_execution_core/ansible_runner'
-  require 'foreman_ansible_core/remote_execution_core/settings_override'
+  require 'foreman_ansible_core/playbook_task_launcher'
   require 'foreman_ansible_core/ansible_runner_task_launcher'
-  ForemanRemoteExecutionCore::Actions::RunScript.send(
-    :prepend,
-    ForemanAnsibleCore::RemoteExecutionCore::SettingsOverride
-  )
+
+  if defined?(SmartProxyDynflowCore)
+    SmartProxyDynflowCore::TaskLauncherRegistry.register('ansible-runner',
+                                                         ForemanAnsibleCore::AnsibleRunnerTaskLauncher)
+    SmartProxyDynflowCore::TaskLauncherRegistry.register('ansible-playbook',
```

Comment:
@adamruzicka: I've changed the way how we hook in into the rex, so that we don't need to do the monkey-patching. As a result, the async_ssh setting doesn't influence the playbook runner (that was not the case, as the async_ssh was causing the refresh intervals to be much longer by default)

Reply:
Can we do this? Wouldn't this break running with older proxies?

Reply:
yes